sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -243,13 +243,20 @@ case class DataSource(
         val path = caseInsensitiveOptions.getOrElse("path", {
           throw new IllegalArgumentException("'path' is not specified")
         })
+        // If we're reading files from outputs of another streaming query, then it does not make
+        // sense to glob files since we would get files from the metadata log.
+        // Thus we would figure out whether there exists some metadata log only when user gives a
+        // non-glob path.
+        val sourceHasMetadata =
+          !SparkHadoopUtil.get.isGlobPath(new Path(path)) && hasMetadata(Seq(path))
```

Comment:
I guess `sourceHasMetadata` is generated here is because of `hasMetadata`. Could you move `hasMetadata` to `object FileStreamSink`? Then you can do it inside `FileStreamSource`.